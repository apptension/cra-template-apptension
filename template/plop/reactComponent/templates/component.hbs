import React from 'react';
import PropTypes from 'prop-types';

import { Container } from './{{ camelCase name }}.styles';

export const {{ pascalCase name }} = () => {
  return <Container>{{ pascalCase name }} component</Container>;
};

{{ pascalCase name }}.propTypes = {};
