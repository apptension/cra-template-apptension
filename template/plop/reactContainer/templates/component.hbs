import React from 'react';
import PropTypes from 'prop-types';

import { Container } from './{{ camelCase name }}.styles';

export const {{ pascalCase name }}Component = () => {
  return <Container>{{ pascalCase name }} component</Container>;
};

{{ pascalCase name }}Component.propTypes = {};
