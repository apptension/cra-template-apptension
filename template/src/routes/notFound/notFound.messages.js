/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  pageTitle: {
    id: 'notFound.pageTitle',
    defaultMessage: 'Not found',
  },
  title: {
    id: 'notFound.title',
    defaultMessage: 'Error: 404',
  },
});
