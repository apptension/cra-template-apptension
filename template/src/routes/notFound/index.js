import { hot } from 'react-hot-loader/root';
import { NotFoundComponent } from './notFound.component';

export const NotFound = hot(NotFoundComponent);
