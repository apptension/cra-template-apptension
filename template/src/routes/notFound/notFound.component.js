import React from 'react';
import { Helmet } from 'react-helmet-async';
import { FormattedMessage, useIntl } from 'react-intl';

import messages from './notFound.messages';
import { Container } from './notFound.styles';
import { H1 } from '../../theme/typography';

export const NotFoundComponent = () => {
  const { formatMessage } = useIntl();

  return (
    <Container>
      <Helmet title={formatMessage(messages.pageTitle)} />

      <H1>
        <FormattedMessage {...messages.title} />
      </H1>
    </Container>
  );
};
