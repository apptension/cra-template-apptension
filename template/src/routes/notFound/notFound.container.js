import React from 'react';
import { useIntl } from 'react-intl';

import { NotFoundComponent } from './notFound.component';

export const NotFoundContainer = () => {
  const { formatMessage } = useIntl();

  const componentProps = {
    formatMessage,
  };

  return <NotFoundComponent {...componentProps} />;
};
