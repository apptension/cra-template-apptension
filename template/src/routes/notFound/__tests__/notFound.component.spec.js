import React from 'react';
import { shallow } from 'enzyme';

import { NotFound } from '../notFound.component';

describe('NotFound: Component', () => {
  const defaultProps = {};

  const component = props => <NotFound {...defaultProps} {...props} />;

  it('should render correctly', () => {
    const wrapper = shallow(component());
    expect(wrapper).toMatchSnapshot();
  });
});
