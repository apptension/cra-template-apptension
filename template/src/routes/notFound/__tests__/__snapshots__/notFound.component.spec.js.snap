// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound: Component should render correctly 1`] = `
<styled.div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="notFound.pageTitle / Not found / {}"
  />
  <styled.h1>
    <FormattedMessage
      defaultMessage="Error: 404"
      id="notFound.title"
    />
  </styled.h1>
</styled.div>
`;
